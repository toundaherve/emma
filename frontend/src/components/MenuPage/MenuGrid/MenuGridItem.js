import React from "react";

const MenuGridItem = () => {
  return (
    <div className="menu-grid-item js-menu-grid-item">
      <div
        className="grid-item"
        itemScope=""
        itemType="https://schema.org/Product"
      >
        <meta itemProp="brand" content="Chili's" />
        <meta itemProp="category" content="Appetizers" />
        <meta
          itemProp="image"
          content="https://static.olocdn.net/menu/chilis/a30644e222a6f6d261f13b5bf1f0b089.jpg"
        />
        <div className="item-image-placeholder"></div>
        <a
          id="menu-item-image-P101942"
          className="item-image b-lazy b-loaded"
          href="/menu/appetizers/triple-dipper"
          title="Triple Dipper™"
          style={{
            backgroundImage: `url("https://static.olocdn.net/menu/chilis/a30644e222a6f6d261f13b5bf1f0b089.jpg")`,
          }}
        >
          Triple Dipper™
        </a>
        <div className="item-body">
          <a
            id="menu-item-name-P101942"
            className="item-title js-item-title height-calculated"
            href="/menu/appetizers/triple-dipper"
            itemProp="url"
            style={{ height: "25px" }}
          >
            <span itemProp="name">Triple Dipper™</span>
          </a>
          <div className="item-description-wrapper">
            <div
              className="item-description js-item-description height-calculated"
              style={{ height: "70px" }}
            >
              <span
                itemProp="description"
                data-js-extra-max-height="3"
                title="Why choose one when you can choose three? Select three appetizers and enjoy! Served with dipping sauces."
              >
                Why choose one when you can choose three? Select three
                appetizers and enjoy! Served with dipping sauces.
              </span>
            </div>
          </div>
          <a
            id="menu-item-button-P101942"
            className="btn slim btn-menu-item"
            href="/menu/appetizers/triple-dipper"
          >
            Order
          </a>
        </div>
      </div>
    </div>
  );
};

export default MenuGridItem;
