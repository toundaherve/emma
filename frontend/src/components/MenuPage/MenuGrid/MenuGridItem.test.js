import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";

import MenuGridItem from "./MenuGridItem";

it("renders correctly", () => {
  const component = renderer.create(
    <Router>
      <MenuGridItem />
    </Router>
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
