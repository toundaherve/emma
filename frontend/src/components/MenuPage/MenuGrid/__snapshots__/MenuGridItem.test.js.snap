// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="menu-grid-item js-menu-grid-item"
>
  <div
    className="grid-item"
    itemScope=""
    itemType="https://schema.org/Product"
  >
    <meta
      content="Chili's"
      itemProp="brand"
    />
    <meta
      content="Appetizers"
      itemProp="category"
    />
    <meta
      content="https://static.olocdn.net/menu/chilis/a30644e222a6f6d261f13b5bf1f0b089.jpg"
      itemProp="image"
    />
    <div
      className="item-image-placeholder"
    />
    <a
      className="item-image b-lazy b-loaded"
      href="/item"
      id="menu-item-image-P101942"
      onClick={[Function]}
      style={
        Object {
          "backgroundImage": "url(\\"https://static.olocdn.net/menu/chilis/a30644e222a6f6d261f13b5bf1f0b089.jpg\\")",
        }
      }
      title="Triple Dipper™"
    >
      Triple Dipper™
    </a>
    <div
      className="item-body"
    >
      <a
        className="item-title js-item-title height-calculated"
        href="/item"
        id="menu-item-name-P101942"
        itemProp="url"
        onClick={[Function]}
        style={
          Object {
            "height": "25px",
          }
        }
      >
        <span
          itemProp="name"
        >
          Triple Dipper™
        </span>
      </a>
      <div
        className="item-description-wrapper"
      >
        <div
          className="item-description js-item-description height-calculated"
          style={
            Object {
              "height": "70px",
            }
          }
        >
          <span
            data-js-extra-max-height="3"
            itemProp="description"
            title="Why choose one when you can choose three? Select three appetizers and enjoy! Served with dipping sauces."
          >
            Why choose one when you can choose three? Select three appetizers and enjoy! Served with dipping sauces.
          </span>
        </div>
      </div>
      <a
        className="btn slim btn-menu-item"
        href="/item"
        id="menu-item-button-P101942"
        onClick={[Function]}
      >
        Order
      </a>
    </div>
  </div>
</div>
`;
